code/datums/elements/hatstacker.dm

```diff
@@ -0,0 +1,130 @@
+/**
+ * ### hat stacker element!
+ *
+ * Bespoke element (1-per-unique-argument in existence) that lets helmets stack hats on top of themselves!
+ */
+/datum/element/hatstacker
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+	///Whitelist of attachable hats, supplied as argument 2
+	var/list/attachable_hats_list
+
+/datum/element/hatstacker/Attach(datum/target, list/hat_whitelist)
+	. = ..()
+	if(!hat_whitelist)
+		attachable_hats_list = subtypesof(/obj/item/clothing/head)	//PLACEHOLDER - TODO - Add default list of stackable hats
+	attachable_hats_list = hat_whitelist
```

Comment:
the order you do this is in seems wrong

Reply:
it should probably be an if-else, its more just in case the element is added without the second variable; right now it WILL indeed set the list to null if there's no hat_whitelist, because I did, indeed, write it wrong